tests/CMakeLists.txt

```diff
@@ -54,7 +54,10 @@ else ()
   endif()
 endif ()
 
-add_subdirectory(core_tests)
+# Skip the core_tests if we are running in Travis-CI because they will take too long
+if (NOT DEFINED ENV{TRAVIS})
```

Comment:
Is this special-case necessary, given the later special-case around add_custom_target? If not, I would remove it, to minimize the special-cases and also to leave open the option to manually invoke the target to build this test (the extra time for cmake to parse that subdir is tiny).